stimela/backends/runner.py

```diff
@@ -60,7 +75,10 @@ def validate_backend_settings(backend_opts: Dict[str, Any], log: logging.Logger)
                 backend_name = name
                 break
     else:
-        raise BackendError(f"selected backends ({', '.join(selected)}) not available")
+        raise BackendError(
+            f"selected backends ({', '.join(selected)}) not available, "
+            f"or the cab '{cab.name}' does not specify a container image"
```

Comment:
This will fail if cab is None.

Reply:
We definitely need to consolidate these backend schemas/modules because it's quite confusing  and has a lot of duplication.